src/User/User.php

```diff
@@ -179,6 +228,12 @@ public function logout($destroySession = true): bool
             if ($destroySession && $this->session) {
                 $this->session->destroy();
             }
+
+            // Remove the cookie
```

Comment:
This could be added to a method in the Cookie class i.e.

```php
class Cookie
{
    // ...
    public function remove()
    {
        $new = clone $this;
        $new->expire = 1;
        return $new;
    }
```

Reply:
With this I think we have a problem, because the expiredAt parameters is Datetime. Is it ok this?. By the way, I have a question related to this implementation: Why do they clone, what advantages does it have?

Reply:
`clone` gives you immutability that ensures that there will be no side effects such as:

```php
$cookie = new Cookie();

$cookieToAdd = new CookieCollection();
$cookieToAdd($cookie);

// if in saveCookie() we modify the object, it will end up modified in collection
// usually that's not what we want
$anotherClass->modifyAndSaveCookie($cookie);

$andAnotherClass->sendCookies($cookie);
```

Reply:
Cookies are going to be adjusted: https://github.com/yiisoft/yii-web/pull/244

Overall the idea with `remove()` is about setting date in the past. In the pull request mentioned there's such a method called `expire()`.

Reply:
@samdark ahh ok thanks for explaining. 

Regarding the elimination of cookies, we hope to merge that PR, right?. I updated this PR assuming the expired method exists. But I don't sure if is correctly.

Reply:
Yes, it will be merged.

Reply:
@samdark Ok, I understand that this PR would be already, right?